app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -357,6 +359,39 @@ public boolean onQueryTextChange(String text) {
         updateUndoRedoIconStates();
     }
 
+    @Override
+    public boolean onReceiveKeyPress(int keyCode, KeyEvent event) {
+        if (event.isCtrlPressed()) {
+            if (event.isShiftPressed() && keyCode == KeyEvent.KEYCODE_Z) {
+                if (_editTextUndoRedoHelper != null && _editTextUndoRedoHelper.getCanRedo()) {
+                    _hlEditor.withAutoFormatDisabled(_editTextUndoRedoHelper::redo);
+                    updateUndoRedoIconStates();
+                }
+                return true;
+            } else if (keyCode == KeyEvent.KEYCODE_B) {
+                _format.getActions().onActionClick(R.string.abid_markdown_bold);
```

Comment:
This will not work with other formats. IMO we should do one of the following:

Either
Option 1: Rename all format specific but actually common action strings to common values
i.e. `abid_markdown_bold`, `abid_wikitext_bold` etc all to `abid_bold`

Or

Option 2: Create actionBold(), actionItalics() etc in ActionButtonBase and override them in each type specific <Format>ActionButtons

Reply:
I think the option 2 may be better.